packages/runtime/test/integration/updateApp.ts

```diff
@@ -172,4 +171,144 @@ describe("Algorand Smart Contracts - Update Application", function () {
 	});
+
+	describe("Extra Pages", function () {
+		useFixture("stateful");
+		const storageConfig = {
+			appName: "app",
+			globalBytes: 1,
+			globalInts: 1,
+			localBytes: 1,
+			localInts: 1,
+		};
+
+		this.beforeAll(function () {
+			oldApprovalProgramFileName = "counter-approval.teal";
+			clearProgramFilename = "clear.teal";
+		});
+
+		function deployApp(approvalProgramFilename: string, clearProgramFilename: string, appName: string) {
+			const appID = runtime.deployApp(
+				creator.account,
+				{
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: approvalProgramFilename,
+					clearProgramFilename: clearProgramFilename,
+					...storageConfig,
+					appName: appName,
+				},
+				{}
+			).appID;
+			return appID;
+		}
+
+		function updateApp(approvalProgramFilename: string, clearProgramFilename: string, appName: string, extraPages: number = 0) {
+			runtime.updateApp(
+				appName,
+				creator.address,
+				appID,
+				{
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: approvalProgramFilename,
+					clearProgramFilename: clearProgramFilename,
+					extraPages: extraPages
+				},
+				{},
+				{}
+			)
+		}
+
+		it("Should pass when updated program length does not exceed total allowed program length", function () {
+			// create app
+			appID = deployApp(oldApprovalProgramFileName, clearProgramFilename, "app1");
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			let app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			assert.doesNotThrow(
+				() => updateApp("counter-approval.teal", "clear.teal", "app1")
+			);
+
+			// verify updated app
+			assert.isDefined(runtime.getApp(appID));
+		});
+
+		it("Should fail when updated program length exceeds total allowed program length", function () {
+			// create app
+			appID = deployApp(oldApprovalProgramFileName, clearProgramFilename, "app2");
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			const app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			expectRuntimeError(
+				() => updateApp("very-long-approval.teal", "clear.teal", "app2"),
+				RUNTIME_ERRORS.TEAL.MAX_LEN_EXCEEDED
+			);
+		});
+
+		it("Should fail when no extra pages were defined for large approval program", function () {
+			// create app
+			appID = deployApp(oldApprovalProgramFileName, clearProgramFilename, "app3");
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			const app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			expectRuntimeError(
+				() => updateApp("very-long-approval.teal", "clear.teal", "app3"),
+				RUNTIME_ERRORS.TEAL.MAX_LEN_EXCEEDED
+			);
+		});
+
+		it("Should pass when sufficient extra pages was defined", function () {
+			// create app
+			appID = deployApp(oldApprovalProgramFileName, clearProgramFilename, "app4");
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			const app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			// should pass because total 2 pages needed
```

Comment:
because one extra page is needed? I think it would better much the test case where we pass extraPages=1

Reply:
Yes. Extra pages are needed.

Reply:
Yes.